### PR TITLE
Implements new SMS Provider ASPSMS.com

### LINKED ISSUE #11213: SMS Provider: new integration for ASP SMS
Goal of this issue is to integrate a new SMS API Provider in Piwik. Currently Clockwork SMS provider is available but we'd like to introduce a new Swiss-based provider: ASP SMS. https://www.aspsms.com/

### How to call the API to send SMS

* we could simply use this super simple API: https://www.aspsms.com/en/http/
* or maybe we could use this existing PHP library which connects to the API? https://github.com/nadar/aspsms 

### How to get counted as affiliate when users send SMS

* Set "AffiliateID" parameter to "227830" when calling operations that send text messages. 
*  use this link when linking to ASPSMS create account for our users:
http://www.aspsms.com/en/?REF=227830

#### What are ASPSMS Credits and how do I buy some?

For sending an SMS, you need so-called ASPSMS credits,
which are purchased in advance. The ASPSMS credits do
not expire. Payment by bank transfer, various credit
cards such as Eurocard/Mastercard, Visa, American
Express or Diners Club, PayPal or Swiss Postcard.
https://www.aspsms.com/instruction/payment.asp

### PR DESCRIPTION
**Pull request by sgiehl**
fixes #11213

### CODE REVIEW

File: plugins/MobileMessaging/SMSProvider/Clockwork.php [hunk 1]
```diff
@@ -53,14 +53,25 @@ public function getDescription()
 			';
     }
 
-    public function verifyCredential($apiKey)
+    public function getCredentialFields()
     {
-        $this->getCreditLeft($apiKey);
+        return array(
```
Comment (tsteur) [inline plugins/MobileMessaging/SMSProvider/Clockwork.php]: Can we move this as default to SMS Provider base class? Then we likely wont break other plugins like http://plugins.piwik.org/FreeMobileMessaging

Comment (sgiehl) [inline plugins/MobileMessaging/SMSProvider/Clockwork.php]: good point. Will change that.

**Comment by mattab** [comment]
Feedback: 

* It's not possible to click "Save" button in the Mobile Messaging > Update page. Save button is disabled. Reproduced when there was already Clockwork recorded and i'm trying to change it to aspsms.

disabled

**Comment by sgiehl** [comment]
Hm. that used to work. I'll have a look

**Comment by sgiehl** [comment]
@mattab should hopefully be fixed now :)

**Comment by mattab** [comment]
Works. We now have a new SMS Provider, awesome!

**Comment by mattab** [comment]
Feedback to apply @sgiehl 

- Username
Should be: Userkey

- Your clockwork SMS API account is....
Should be: Your ASPSMS account is....


- First, get an account at ASP SMS
Should be: First, get an account at ASPSMS


- ...over 600 networks
Should be:... over 900 networks

- ...starts from 0.09 USD (0.06 EUR).
Should be: ...starts from 0.06 USD (0.04 EUR)

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ The Product Changelog at **[piwik.org/changelog](http://piwik.org/changelog)** l
 
 ## Piwik 3.0.2
 
+### New Features
+* A new SMS provider for sms reports has been added: [ASPSMS.com](http://www.aspsms.com/en/?REF=227830)
+
+### Breaking Changes
+* SMS provider now need to define their credential fields by overwriting `getCredentialFields()`. This allows to have SMS providers that require more than only an API key.
```
Comment (tsteur) [inline CHANGELOG.md]: Is this still a breaking change? Or did we make it backwards compatible? Would be really great to keep BC and if not release 3.1. But ideally I would really want to not break BC in 3.X in general when it is possible to keep BC

Comment (sgiehl) [inline CHANGELOG.md]: good point. I'll change that. It's not breaking BC anymore

Comment (tsteur) [inline CHANGELOG.md]: 👍

**Pull request merged**